### ISSUE TITLE
[Feature] Upgrade Armeria apm-sdk-plugin from 0.98.x to 1.21.x

### ISSUE THREAD

**Issue by zhangrm** [apache/skywalking#10278]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

The latest version supported by the apm-sdk-plugin of Skywalking is 0.98. Upgrade the apm-sdk-plugin to armeria-1.21.x.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
The plugin is designed for specific versions. Basically, when we face incompatible new version, we should create a new plugin for newer versions, rather than bumping up version.

**Comment by wu-sheng** [comment]
https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/java-plugin-development-guide/#implement-plugin

Witness class and methods could be used to help plugin identifying versions.

> Set up witnessClasses and/or witnessMethods if the instrumentation has to be activated in specific versions.

**Comment by zhangrm** [comment]
OK，I'm sorry I'm not accurate, I mean by creating a new version plugin to adapt to the new version.

**Comment by zhangrm** [comment]
For maven module management under the apm-sdk-plugin directory, I have the following ideas:

1. SDK main module naming format: apm - ${sdk_name} - plugins, where ${sdk_name} is the real name of the sdk, for example: apm-armeria-plugins

2. The naming format of the sub-module under the main module of the SDK: apm - ${sdk_name} - ${version} - plugins, where ${sdk_name} is the real name of the sdk, and ${version} is the adaptive version of the SDK, for example: apm-armeria-0.99.x-plugin

3. The directory name of Maven module is consistent with<artifactId>and<name>in pom.xml.

To sum up, take "armeria" as an example to show the structure of maven module as shown below:

<img width="494" alt="image" src="https://user-images.githubusercontent.com/6973089/213262732-f1efb56d-2a9a-4ac6-8bfe-afdbf66680d0.png">

**Comment by wu-sheng** [comment]
I think this is not an issue about changing the folder structure. You are better to submit a pull request to add `apm-armeria-plugins` and move the existing two into it, then send another pull request for the new feature.

**Comment by zhangrm** [comment]
That's a good idea.

**Comment by wu-sheng** [comment]
Are you still working on this?

**Comment by zhangrm** [comment]
Yes, I will submit it this weekend

**Comment by wu-sheng** [comment]
@zhangrm New feature has been merged. Please follow up the discussion, and move the existing two plugins into the new created folder.

**Comment by zhangrm** [comment]
This issue has been resolved
